### PR TITLE
Added env based support for HTTPS connection

### PR DESCRIPTION
**Pull request by Saranya-jena**
## Proposed changes

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: chaoscenter/authentication/api/main.go [hunk 1]
```diff
@@ -114,7 +116,21 @@ func main() {
...
+	enableHTTPSConnection, err := strconv.ParseBool(utils.EnableHTTPSConnection)
+	if err != nil {
+		log.Errorf("unable to parse boolean value %v", err)
+	}
+
+	if enableHTTPSConnection {
+		if utils.CustomTlsCert != "" && utils.TlSKey != "" {
+			go runGrpcServerWithTLS(applicationService)
+		} else {
+			log.Fatalf("Failure to start chaoscenter authentication REST server due to empty TLS cert file path and TLS key path")
```
Comment (ispeakc0de) [inline chaoscenter/authentication/api/main.go:128]:
```suggestion
			log.Fatalf("Failure to start chaoscenter authentication GRPC server due to empty TLS cert file path and TLS key path")
```

**Review by ispeakc0de — APPROVED** [review]

**Review by SarthakJain26 — APPROVED** [review]

File: chaoscenter/authentication/api/main.go [hunk 2]
```diff
@@ -172,10 +188,27 @@ func runRestServer(applicationService services.ApplicationService) {
...
-	log.Infof("Listening and serving HTTP on %s", utils.Port)
-	err := app.Run(utils.Port)
+	enableHTTPSConnection, err := strconv.ParseBool(utils.EnableHTTPSConnection)
 	if err != nil {
-		log.Fatalf("Failure to start litmus-portal authentication REST server due to %v", err)
+		log.Errorf("unable to parse boolean value %v", err)
+	}
+
+	if enableHTTPSConnection {
+		if utils.CustomTlsCert != "" && utils.TlSKey != "" {
```
Comment (Jonsy13) [inline chaoscenter/authentication/api/main.go:197]: We can rename it wrt `paths`

**Review by namkyu1999 — APPROVED** [review]
can you check `build-pipeline / backend-checks (pull_request) `?

File: chaoscenter/authentication/api/main.go [hunk 3]
```diff
@@ -195,3 +237,31 @@ func runGrpcServer(applicationService services.ApplicationService) {
...
+	grpcServer := grpc.NewServer(grpc.Creds(tlsCredentials))
+
+	grpcApplicationServer := grpcHandler.ServerGrpc{ApplicationService: applicationService}
+
+	grpcPresenter.RegisterAuthRpcServiceServer(grpcServer, &grpcApplicationServer)
+
+	log.Infof("Listening and serving gRPC on %s with TLS", utils.GrpcPort)
+	err = grpcServer.Serve(lis)
+	if err != nil {
+		log.Fatalf("Failure to start litmus-portal authentication GRPC server due to %v", err)
```
Comment (Jonsy13) [inline chaoscenter/authentication/api/main.go:265]:
```suggestion
		log.Fatalf("Failure to start chaos-center authentication GRPC server due to %v", err)
```

File: chaoscenter/authentication/api/main.go [hunk 4]
```diff
@@ -195,3 +237,31 @@ func runGrpcServer(applicationService services.ApplicationService) {
...
+
+func runGrpcServerWithTLS(applicationService services.ApplicationService) {
+
+	// Starting gRPC server
+	lis, err := net.Listen("tcp", utils.GrpcPortHttps)
+	if err != nil {
+		log.Fatalf("Failure to start litmus-portal authentication server due to %s", err)
+	}
+
+	// configuration of the certificate what we want
```
Comment (Jonsy13) [inline chaoscenter/authentication/api/main.go:249]:
```suggestion
	// configuring TLS config based on provided certificates & keys
```

File: chaoscenter/graphql/server/pkg/authorization/validate.go [hunk 5]
```diff
@@ -20,6 +21,7 @@ func ValidateRole(ctx context.Context, projectID string,
 		requiredRoles,
 		invitation)
 	if err != nil {
+		fmt.Println("errrrrrrrrrrrr ", err)
```
Comment (Jonsy13) [inline chaoscenter/graphql/server/pkg/authorization/validate.go:24]: This can be removed

File: chaoscenter/graphql/server/server.go [hunk 6]
```diff
@@ -106,7 +108,20 @@ func main() {
...
+
+	enableHTTPSConnection, err := strconv.ParseBool(utils.Config.EnableHTTPSConnection)
+	if err != nil {
+		logrus.Errorf("unable to parse boolean value %v", err)
+	}
+
 	go startGRPCServer(utils.Config.RpcPort, mongodbOperator) // start GRPC serve
+	if enableHTTPSConnection {
+		if utils.Config.ServerTlsCertPath != "" && utils.Config.ServerTlsKeyPath != "" {
+			go startGRPCServerWithTLS("8001", mongodbOperator) // start GRPC serve
```
Comment (Jonsy13) [inline chaoscenter/graphql/server/server.go:120]: Port is hardcoded

File: chaoscenter/graphql/server/server.go [hunk 7]
```diff
@@ -148,8 +163,28 @@ func main() {
...
-	log.Fatal(http.ListenAndServe(":"+utils.Config.HttpPort, router))
+	log.Infof("graphql server running at http://localhost:%s", utils.Config.HttpPort)
+	go func() {
+		err := http.ListenAndServe(":"+utils.Config.HttpPort, router)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+	}()
+	if enableHTTPSConnection {
+		// configuration of the certificate what we want
```
Comment (Jonsy13) [inline chaoscenter/graphql/server/server.go:174]: do we need log here as well same as above

`log.Infof("graphql server running at https://localhost:%s", utils.Config.HttpsPort)`

**Review by Jonsy13 — APPROVED** [review]

**Pull request merged**